yaqd_andor/_andorsdk2_ixon.py

```diff
@@ -0,0 +1,309 @@
+__all__ = ["AndorSdk2Ixon"]
+
+import asyncio
+import numpy as np
+import yaqc  # type: ignore
+
+from yaqd_core import IsDaemon, IsSensor, HasMeasureTrigger, HasMapping, HasDependents
+from typing import Dict, Any, List, Union
+from . import _andor_sdk2
+import ctypes
+from time import sleep
+
+
+class AndorSdk2Ixon(_andor_sdk2.AndorSDK2):
+    _kind = "andorsdk2-ixon"
+
+    def __init__(self, name, config, config_filepath):
+        super().__init__(name, config, config_filepath)
+
+        self.stop_update == False
+        self._spec_position = self._config["spec_position"]
+        self.exposure_time = self._state["exposure_time"]
+        self._channel_names = ["image"]
+        self._channel_units = {"image": "counts"}
+        self.sensor_width = self._config["sensor_width"]
+        self.sensor_height = self._config["sensor_height"]
+        self.max_width = self._config["pixel_width"]
+        self.max_height = self._config["pixel_height"]
+        self.preamp_gain = int(self._config["simple_preamp_gain_control"])
+
+        if isinstance(self._spec_position, str):
+            host, port = self._spec_position.split(":")
+            self.has_mono = True
+            self.spec_client = yaqc.Client(int(port), host=host)
+        elif isinstance(self._spec_position, float):
+            self.has_mono = True
+            self.spec_client = None
+        else:
+            self.has_mono = False
+            self.spec_client = None
+            self._spec_position = None
+
+        hw = self.get_dependent_hardware()
+
+        # find devices
+        code, device_count = self.sdk.GetAvailableCameras()
+        if device_count == 0:
+            raise ConnectionError("No devices found.")
+        if code != int(20002):
+            self.logger.debug(f"getavailablecameras error {str(self.errorlookup(code))}")
+
+        for i in range(device_count):
+            ret, serial = self.sdk.GetCameraSerialNumber()
+            if int(serial) == int(self._config["serial_number"]):
+                print("    Serial No   : ", serial)
+                self.hndl = self.sdk.GetCameraHandle(i)
+                break
+        else:
+            raise ConnectionError(
+                r"device with serial number {0} not found".format(self._config["serial_number"])
+            )
+
+        self._set_aoi()
+        self._initialize_spec_settings()
+        self._gen_mappings()
+
+        # see p97 of SDK2 manual on the conversion of noise filters from sdk3 to sdk2 or v-v.
+        # Commented out until further work done to establish differences between
+        # these filters and SDK3.
+        # self.sdk.Filter_SetMode(int(self._config["spurious_noise_filter"]))
+        # self.sdk.Filter_SetThreshold(float(self._config["filter_threshold"]))
+
+        self.sdk.SetPreAmpGain(self.preamp_gain)
+        self.logger.info(f"PreAmpGain: {self.preamp_gain}")
```

Comment:
Consider introducing gain as a yaq property (like you did with exposure time), so the client has setter and getter capabilities.  Here, the only way one can verify the gain is to watch the log.

Reply:
I think I'll pass on employing as a property this round, maybe introduce it with the noise filters if users need to modify these values on-the-fly.